Robust/ViewModel/ProductViewModel.cs

```diff
@@ -14,6 +16,73 @@ internal class ProductViewModel : ViewModelBase
     private ProductRepository _repository = new();
     private ObservableCollection<Product> _products = [];
 
+    //This Collection is bound to the combobox which displays all the different categories.
+    public ObservableCollection<Category> Categories { get; set; }
+    //This Collection is used for adding products to the MainWindow
+    public ObservableCollection<Product> Products { get; set; }
+    //This Collection contains the products that are to be displayed in the MainWindow.
+    public ObservableCollection<Product> SelectedProducts { get; set; }
+
+    //This property is used to select images based on their category.
+    private Category _selectedCategory;
+
+    public Category SelectedCategory
+    {
+        get { return _selectedCategory; }
+        set 
+        { 
+            _selectedCategory = value;
+            OnPropertyChanged();
+            DisplayProducts();
+            CheckDisplayAll();
+            CheckAvailability();            
```

Comment:
Vi kalder 4 metoder hver gang der er en ny value der sættes, måske vi kan lave en hovedfunktion som gør kan kalde de forskellige metoder? Så bliver det nemmere hver gang vi skal ind og finde alle metoderne der har noget med "DisplayProducts"

Dette kunne f.eks. bare være

DisplayProducts -> Check om det er en kategori -> Kald DisplayCategoryProducts ellers Kald DisplayAllCategories

Dette vil også gøre det nemmere for en unit test der umiddelbart bare skal kalde 1 funktion istedet for 4.